### PR TITLE
AWS AppSync Support

### PR DESCRIPTION
**Pull request by bjorg**
The following changes allow me to use GraphQL.Client from a Blazor WebAssembly app. Please review and let me know what I need to address before submitting a pull-request.

**Changes:**
1. Added `WebSocketEndPoint` property to allow specifying a separate websocket endpoint.
2. Added `PreprocessSubscriptionRequest` to disambiguate between query and subscription pre-processing.
3. Converted some `Console.WriteLine()` statements to `Debug.WriteLine()`.
4. Added some `Debug.WriteLine()` to help with debugging the AppSync protocol.
5. Fixed an issue were WebAssembly was not properly detected (oops).

### CODE REVIEW

File: src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs
Comment (bjorg) [inline src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs:201]: Technically, the client should be waiting for the `connection_ack` response before proceeding. For now, I put a TODO comment here as I don't have the experience with RX to know how to properly codify this step.

**Comment by bjorg** [comment]
Here a sample Blazor app built with these changes: https://github.com/bjorg/GraphQlAppSyncTest/blob/main/MyApp/Pages/Index.razor

Using the new subscription pre-processor, I was able create a fairly simple adapter to configure GraphQL.Client for AppSync:
https://github.com/bjorg/GraphQlAppSyncTest/blob/main/MyApp/GraphQLHttpClientOptionsEx.cs

**Comment by bjorg** [comment]
@rose-a Do you have any feedback on these changes?

**Comment by rose-a** [comment]
@bjorg Sorry, I did not yet find time to review your work, I hope I'll manage sometime this week...

**Review by rose-a — APPROVED** [review]

**Comment by jgowdy** [comment]
This would be very valuable for my team.  Thanks for contributing this!

**Comment by bjorg** [comment]
I've started splitting these changes up into smaller, bitesize pull-requests.
1. Fix Blazor WASM check: https://github.com/graphql-dotnet/graphql-client/pull/289
2. Debug logging for received payloads: https://github.com/graphql-dotnet/graphql-client/pull/290
3. Fix issues with `GraphQLRequest`: https://github.com/graphql-dotnet/graphql-client/pull/291

Once they are approved and merged, I will update this pull-request. Thx.

**Comment by Shane32** [comment]
👍 It's always easier to review smaller PRs 👍

File: src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs
Comment (rose-a) [reply src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs:201]: In the [Apollo GraphQL over WebSocket Protocol description](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md#gql_connection_ack) they write that the server **may** respond with `GQL_CONNECTION_ACK`... so it might be better to not wait for it?

Comment (bjorg) [reply src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs:201]: Urgh... _may_ is the worst.

**Comment by MikePapinski** [comment]
as the changes has been merged, can we close this PR to do not mislead people that AWS AppSync is not working with this client?

> I've started splitting these changes up into smaller, bitesize pull-requests.
> 
> 1. Fix Blazor WASM check: [fix Blazor WASM check #289](https://github.com/graphql-dotnet/graphql-client/pull/289)
> 2. Debug logging for received payloads: [Debug logging for received payloads #290](https://github.com/graphql-dotnet/graphql-client/pull/290)
> 3. Fix issues with `GraphQLRequest`: [GraphQLRequest fixes #291](https://github.com/graphql-dotnet/graphql-client/pull/291)
> 
> Once they are approved and merged, I will update this pull-request. Thx.

**Comment by rose-a** [comment]
Closing as beeing outdated...

**Pull request closed without merging**
